base/MAPL_TimeMethods.F90

```diff
@@ -44,6 +52,7 @@ function new_time_data(clock,ntime,frequency,offset,funits,integer_time,rc) resu
     logical, optional, intent(in) :: integer_time
     integer, optional, intent(Out) :: rc
 
+    tData%is_initialized = .true.
```

Comment:
Is there a path where one does not go through this constructor?  Otherwise it looks like `is_initialized` is always `.true.` despite having a different default value in the constructor.  I'm probably missing something basic here.